src/charm.py

```diff
@@ -319,6 +324,34 @@ def _get_session_ssl_cipher(self, event: ActionEvent) -> None:
 
         event.set_results({"cipher": cipher})
 
+    def _get_server_certificate(self, event: ActionEvent) -> None:
+        """Get the server certificate."""
+        certificate = "error"
+        if not self._database_config:
+            return event.set_results({"certificate": certificate})
+        try:
+            process = subprocess.run(
+                [
+                    "openssl",
+                    "s_client",
+                    "-starttls",
+                    "mysql",
+                    "-connect",
+                    f"{self._database_config['host']}:{self._database_config['port']}",
+                ],
+                capture_output=True,
```

Comment:
i am not sure if you adding `encoding="utf-8"` will result in a solution for the non utf characters in the certificate

Reply:
it will break the `subprocess.run` call, not capturing nothing